rfcs/0002-upload-from-dropbox/text.md

```diff
@@ -307,6 +338,142 @@ Checks the status of an existing upload job.
 | 401  | If _dataset_id_ does not exist, the user does not own the collection associated with the dataset, or upload is in-progress. |
 | 400  | If the parameters supplied are invalid.                                                                                     |
 
+##### Status Enumerations
+The statuses in the 200 response are enums defined in the [database](#database-schema):
+
+**upload_status**
+| Values         | Description                                                                                 |
+| -------------- | ------------------------------------------------------------------------------------------- |
+| Waiting        | The upload is enqueued, and waiting for the upload container.                               |
+| Uploading      | The file is actively being uploaded.                                                        |
+| Uploaded       | The upload was completed successfully.                                                      |
+| Failed         | The upload has failed. A new link or file must be uploaded. Any upload progress is deleted. |
+| Cancel Pending | The upload is in the process of being canceled.                                             |
+| Canceled       | The upload has been canceled. Any upload progress is deleted.                               |
+
+**validation_status**
+| Values         | Description                                                                                 |
+| -------------- | ------------------------------------------------------------------------------------------- |
+| Validating     | The validation script is running.                                                           |
+| Valid          | The uploaded file successfully passed validation.                                           |
+| Invalid        | The uploaded file failed validation.                                                         |
+
+**conversion_status**
+| Values         | Description                                                                                 |
+| -------------- | ------------------------------------------------------------------------------------------- |
+| Converting     | The conversion script is running.                                                           |
+| Converted      | Conversion completed and the file was copied to the DP bucket.                              |
+| Failed         | Conversion failed.                                                                          |
+| NA             | Conversion is not being run because this was the uploaded format.                           |
+
+##### Processing Status and State
```

Comment:
Hi @mckinsel !

Sounds like @brianraymor and @maniarathi want the user to have the ability to cancel the process at all stages (upload, validation, and conversion)

So we're wondering if it's possible for BE to delete the dataset row once the cancellation request is accepted (even though the actual cancellation is pending and cleanup will happen asynchronously in the background)? That way if the user refreshed the page, the dataset row won't show up!

Thank you!

CC: @hthomas-czi

Reply:
Hm I'd rather not decide on the exact implementation just yet. When you say "won't show up", do you mean that it won't show up when you list the datasets associated with a collection? We don't have to delete the row to do that, we can just not return datasets where status is in ("Cancel Pending", "Canceled").